File: C++/freecodecamp_c++_course/17_building_a_guessing_game.cpp
#include <iostream>

using namespace std;

int main()
{
  int secretNum = 7;
  int guess;
  int guessCount = 0;
  int maxGuesses = 3;

  while (guess != secretNum) {
    cout << "Guess the secret integer: ";
    cin >> guess;
    guessCount++;

    if (guessCount == maxGuesses) break;
  }

  if (guessCount == 3 && guess != secretNum) {
    cout << "Too many guesses. You lose!" << endl;
  } else {
    cout << "You won!" << endl;
  }
  
  return 0;
}

File: C#/freecodecamp_c#_course/17_building_a_guessing_game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuildingAGuessingGame
{
  class Program
  {
    static void Main(string[] args)
    {
      string secretWord = "shingo";
      string guess = "";
      int guessCount = 0;

      while (guess != secretWord && guessCount < 3)
      {
        Console.Write("Guess the secret word: ");
        guess = Console.ReadLine();

        guessCount++;
      }

      if (guess == secretWord)
        Console.WriteLine("You win!");
      else
        Console.WriteLine("Too many guesses. You lose!");
    }
  }
}
